app/models/user.rb

```diff
@@ -17,4 +19,24 @@ class User < ApplicationRecord
   def liked?(post_id)
     likes_given.where('post_id = ?', post_id).any?
   end
+
+  def pending_requests
+    Friendship.where('friend_id = ? AND confirmed = ?', id, false)
+  end
```

Comment:
- [ ] **`[OPTIONAL]`** We can simplify this with 

`Friendship.where(confirmed: false)` 

Same for `sent_requests`

Reply:
Hey @mosaaleb, this suggestion is not accurate. :thinking: 

Using `Friendship.where('friend_id = ? AND confirmed = ?', id, false)` checks the Friendship table where `friend_id` is the user id. Omitting `friend_id` will return data for both sent and pending request. 

Kindly, review this and get back to us. 

Thanks.

Reply:
You are right :+1: Sorry about this inconvenience.